packages/block-library/src/block/edit.js

```diff
@@ -84,6 +100,24 @@ export default function ReusableBlockEdit( { attributes: { ref }, clientId } ) {
 
 	const blockProps = useBlockProps();
 
+	// state for lock
+	const [ isLocked, setIsLocked ] = useState( true );
+
+	let innerBlocks = <div { ...innerBlocksProps } />;
+	if ( isLocked ) {
+		innerBlocks = <Disabled> { innerBlocks } </Disabled>;
+	}
+
+	const lockContainerClass = isLocked ? 'is-locked' : 'is-unlocked';
+
+	// lock the blocks when deselected
+	useEffect( () => {
+		const isInnerBlock = parentBlockName === 'core/block'; // first check if selectedblock is inner block
```

Comment:
We check if the block is inner block by block type and not by verifying if the specific reusable block id is the parent of the deselected block, if we have multiple reusable blocks on the page will this logic handle that case?

Reply:
Thank you for taking a look 😄 

Yes, it does work when multiple reusable blocks are on the page because first we have to select the reusable block itself, and only after that, we can select the inner blocks. 

So, when multiple reusable blocks are on the page, after working on the first reusable block, we go and select the other one. Now, first, we have to select the outer block (reusable block itself) which won't have a parentBlockName of `core/block` thus it will lock the reusable blocks. The same logic follows in the case of the reusable blocks inside columns.

The bug I came across while testing this was when a reusable block is selected and its inner blocks are not selected right after that but another reusable block is selected.
In such a scenario, the previously selected reusable block remains unlocked. 

I am adding a commit to fix this bug. Once again, thank you for reviewing it! 🙏